### PR TITLE
chore: Addition of npm cache and using npx

### LINKED ISSUE #6374: Add cache for npm
There is no cache for npm in Travis

Also, avoid global installation and use npx instead

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #6374

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,10 +17,11 @@ cache: pip
 
 before-install:
   - npm -g install npm@latest
+  - cache: npm
 
 install:
-  - npm install -g aglio
-  - npm install -g dredd
+  - npx install -g aglio
+  - npx install -g dredd
```
Comment (iamareebjamal) [inline .travis.yml]: Not at all what I meant

File: .travis.yml [hunk 2]
```diff
@@ -17,10 +17,11 @@ cache: pip
 
 before-install:
   - npm -g install npm@latest
+  - cache: npm
```
Comment (iamareebjamal) [inline .travis.yml]: This won't work

File: .travis.yml (hunk 1 above)
Comment (kushthedude) [reply .travis.yml]: Do I need to install `npx` first ?

**Comment by iamareebjamal** [comment]
Read about npx

**Pull request closed without merging**